### DISCUSSION TITLE
OneDrive Version Update and Error Troubleshooting

### DISCUSSION THREAD

**Discussion by csantosbntt** [abraunegg/onedrive#2461] [Q&A] [answered]
Hello everyone,

I am facing an issue with the OneDrive version installed on one of our machines and would appreciate some assistance in resolving it.

On one machine, I am currently using OneDrive version v2.4.9-10-g0bbf162, and it seems to be malfunctioning, displaying the following error message:
```
ERROR: OneDrive returned an error with the following message:
  Error Message: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
Internet connectivity to Microsoft OneDrive service has been restored
ERROR: Unable to query OneDrive to initialize application
```
On another machine, I am running OneDrive version 2.4.23, and it is functioning correctly, allowing me to perform operations without any issues.

I suspect that the problem might be related to the version of OneDrive installed on the first machine. Hence, I am considering updating it to the latest version, which is v2.4.25.

Before proceeding with the update, I have a few questions and concerns:

- Is the error related to the version mismatch? Could the outdated version v2.4.9 be causing the timeout and initialization issue?

- Will updating to the latest version (v2.4.25) likely resolve the error and restore OneDrive's functionality?

- Are there any potential risks or complications involved in updating the version? Could the update potentially break any existing functionalities or configurations?

Additionally, I want to clarify that I am not using "sudo apt-get" to install OneDrive; I have the OneDrive repository directly installed via Git.

If there are any specific steps I should follow to safely update OneDrive to the latest version, kindly provide instructions or references to relevant documentation. I was thinking of using 'git pull', but I am unsure if it's the best approach.

I would greatly appreciate any advice, tips, or suggestions from the community regarding the best course of action. If anyone has encountered a similar issue or has successfully updated their version of OneDrive, please share your insights.

Thank you all for your support and collaboration in resolving this matter.

Best regards,

**Comment by abraunegg** [comment, marked as answer]
@csantosbntt 

> I am facing an issue with the OneDrive version installed on one of our machines and would appreciate some assistance in resolving it

As per the [readme](https://github.com/abraunegg/onedrive#supported-application-version): 

![image](https://github.com/abraunegg/onedrive/assets/4956234/85ced9f7-aeba-4cba-8546-eeff805cb3b7)

You need to upgrade all your systems to v2.4.25

> Additionally, I want to clarify that I am not using "sudo apt-get" to install OneDrive; I have the OneDrive repository directly installed via Git.

If you are using a Debian or Ubuntu based distribution - it is strongly advised you use the OpenSuSE Build Service packages to install. For any other distribution refer to the detailed list here to install your distribution package: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#installing-or-upgrading-using-distribution-packages

> If there are any specific steps I should follow to safely update OneDrive to the latest version, kindly provide instructions or references to relevant documentation. I was thinking of using 'git pull', but I am unsure if it's the best approach.

If you are set on building from source, you *must* follow the instructions for your distribution as per: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#building-from-source---high-level-requirements

> Could the update potentially break any existing functionalities or configurations?

Please review https://github.com/abraunegg/onedrive/releases to understand what is fixed and/or changed in each version so you can determine if any of those changes or fixes will break your existing configuration.
